rpcs3/Emu/RSX/rsx_methods.cpp

```diff
@@ -207,15 +207,15 @@ namespace rsx
 		{
 			if (rsx->in_begin_end)
 			{
-				rsx->append_array_element(arg & 0xFFFF);
-				rsx->append_array_element(arg >> 16);
+				rsx->append_array_element(se_storage<u16>::swap(arg & 0xFFFF));
+				rsx->append_array_element(se_storage<u16>::swap(arg >> 16));
```

Comment:
There's an earlier swap to le long before the method is accessed, which I missed. I also forgot about the swap again when write_index_array_data_to_buffer is called. Thanks for catching that. I'm working blind on this one unfortunately. The swap assumed that arg was in be but it wasn't which is why I misunderstood what you meant by cast to u32.